aws/hll/aws-s3-transfer-manager/src/download/object_meta.rs

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use aws_sdk_s3::operation::{get_object::GetObjectOutput, head_object::HeadObjectOutput};
+
+// TODO(design): how many of these fields should we expose?
+// TODO(docs): Document fields
+
+/// Object metadata other than the body that can be set from either `GetObject` or `HeadObject`
+#[derive(Debug, Clone)]
+pub struct ObjectMetadata {
```

Comment:
Is exposing all of these fields intentional?

Reply:
They are all the common fields between `HeadObject` and `GetObject` and would be public on the response types for those. This is certainly up for design/debate. I would expect though that we expose these somehow as the lowest level download primitive is still "download a single object...just do it intelligently/fast" (i.e. we still need to expose the metadata fields of the object somewhere).